examples/reshare/src/dkg/actor.rs

```diff
@@ -0,0 +1,397 @@
+                                (next_participants, public, Some(share))
+                            }
+                            (next_participants, RoundResult::Polynomial(public)) => {
+                                (next_participants, public, None)
+                            }
+                        };
+
+                        info!(
+                            epoch,
+                            "finalized epoch's reshare; instructing reconfiguration after reshare."
+                        );
+                        let next_epoch = epoch + 1;
+
+                        // Pseudorandomly select some random players to receive shares for the next epoch.
+                        let mut rng = StdRng::seed_from_u64(next_epoch);
+                        let next_players = all_participants
+                            .iter()
+                            .cloned()
+                            .choose_multiple(&mut rng, self.num_participants_per_epoch)
+                            .into_iter()
+                            .collect::<Set<_>>();
+
+                        let transition: EpochTransition<V, C::PublicKey> = EpochTransition {
+                            epoch: next_epoch,
+                            poly: public.clone(),
+                            share: share.clone(),
+                            participants: next_participants.clone(),
+                        };
+                        orchestrator.report(transition).await;
+
+                        // Prune the round metadata for the previous epoch.
+                        self.round_metadata.remove(&epoch.into());
+                        self.round_metadata
+                            .sync()
+                            .await
+                            .expect("metadata must sync");
+
+                        // Rotate the manager to begin a new round.
+                        manager = DkgManager::init(
+                            &mut self.context,
+                            next_epoch,
+                            public,
+                            share,
+                            &mut self.signer,
+                            next_participants,
+                            next_players,
+                            &mut dkg_mux,
+                            &mut self.round_metadata,
+                        )
+                        .await;
+
+                        self.epoch_metadata
+                            .put_sync(EPOCH_METADATA_KEY, next_epoch)
+                            .await
+                            .expect("epoch metadata must update");
+                    };
+
+                    match relative_height.cmp(&(BLOCKS_PER_EPOCH / 2)) {
+                        Ordering::Less => {
+                            // Continuously distribute shares to any players who haven't acknowledged
+                            // receipt yet.
+                            manager.distribute(epoch).await;
```

Comment:
We must take into account the channel rate limit for share distribution or we'll rate limit the processing of other messages between the peers.

Reply:
https://github.com/commonwarexyz/monorepo/pull/1796/commits/631d89e5439d2117aea8dbf0d6d6606369212181